schema.ts

```diff
@@ -837,11 +837,17 @@ export type ObjectSchemaDefinitionType<
 > = ObjectSchemaType<T>;
 
 // This is basically the same as an ObjectSchemaDefinition but includes an Identity that may be injected
-// at upload time (for static schemas) or at runtime (for dynamic schemas). This is the type that is
-// used on the `coda` side for implementations, but should not need to be used by pack makers.
+// at upload time (for static schemas) or at runtime (for dynamic schemas). Pack makers should only need to
+// worry about identity when things outside of a sync table should interact with it, like references or
+// action return values.
 /** @hidden */
 export interface ObjectSchema<K extends string, L extends string> extends ObjectSchemaDefinition<K, L> {
   identity?: Identity;
+
+  /**
+   * Pack makers should never need to interact with this, it's just present for Coda's internal plumbing.
+   */
+  __packId?: number;
```

Comment:
is it intentional that you put two `_`s for this

Reply:
Yeah, but I could be convinced to change it if folks feel strongly.

Reply:
i just haven't seen that before haha is that a practice?

Reply:
A bit, yeah. Like, a programmer might make code that uses a single underscore, so if you're writing code for a programmer to use, you add another layer to avoid potential collisions.

If it's good enough for javascript, it's good enough for me:
![Screen Shot 2022-06-15 at 1 08 50 PM](https://user-images.githubusercontent.com/84040450/173917819-df03d7be-a898-4211-890f-571d21e0fc95.png)

For example, we used it when customizing our json serialization: `var HACK_UNDEFINED_JSON_VALUE = '__CODA_UNDEFINED__';`